File: python/medium/Unique_Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        step_list = []
        for i in range(m):
            step_list.append([])
            for j in range(n):
                if i == 0 or j == 0:
                    step_list[i].append(1)
                else:
                    step_list[i].append(step_list[i][j-1]+step_list[i-1][j])
        return step_list[m-1][n-1]

File: cpp/medium/Unique_Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> grid(m, vector<int>(n, 1));
        for(int j = 1; j < m; ++j) {
            for(int i = 1; i < n; ++i) {
                grid[j][i] = grid[j-1][i] + grid[j][i-1];
            }
        }
        return grid[m-1][n-1];
    }
};
